libsolidity/AST.cpp

```diff
@@ -255,6 +261,37 @@ void StructDefinition::checkRecursion() const
 	}
 }
 
+void EnumDefinition::checkValidityOfMembers() const
+{
+	vector<ASTPointer<EnumValue>> members(getMembers());
+	auto compareDecls = [](ASTPointer<EnumValue> a, ASTPointer<EnumValue> b)
+	{
+		return a->getName() < b->getName();
+	};
+	sort(begin(members), end(members), compareDecls);
+	for (size_t i = 0; i < members.size() - 1; ++i)
+		if (members[i]->getName() == members[i + 1]->getName())
+			BOOST_THROW_EXCEPTION(createTypeError("Duplicate member detected in Enum"));
```

Comment:
Hm, isn't that already detected in `DeclarationRegistrationHelper::registerDeclaration`?